### PR TITLE
Hit api

### PR DESCRIPTION
**Pull request by Haywayaheadshot**
### What?

I created the wireframe for a book store app using react. I also added the components and pages for this app.

### Why?

This is to help users keep track of books they have read with detailed information on attributes of the book.

### How?
- I created the app using react
- I started with the mobile first approach
- I created a hamburger icon for the mobile menu
- I created the components of the app
- I created pages inside the components
- I routed every link to their respective pages
- I moved all styles into one folder in the src directory
- I installed redux and redux toolkit
- I added a redux folder in the src folder
- I created a configureStore file in the redux folder
- I created a books and categories folder in the redux folder. These individual folders contain their respective JavaScript files which contain their respective reducers.
- I added some placeholder books through react-redux functionality
- I used PropTypes for prop validation
- I wrapped my app in Provider in index.js
- I used useDispatch for adding a new book
- I added 'remove' functionality to 'remove' button
- I used useDispatch to display 'under construction'
- I created a new book id inside the book.
- I refactored add and remove books actions.
- I added the API calls using AXIOS library.
- I added a new Slicer to the reducer.
- I added an Object.keys and a map function to receive the correct data from the books.

### CODE REVIEW

File: src/redux/books/Books.js [hunk 1]
```diff
@@ -1,68 +1,59 @@
...
+  name: 'books',
+  initialState,
+  extraReducers: (builder) => {
+    builder.addCase(getBooks.fulfilled, (_, action) => action.payload);
+    builder.addCase(getBooks.rejected, (state) => {
+      const newState = state;
+      newState.status = 'failed';
+    });
+    builder.addCase(getBooks.pending, (_, action) => action.payload);
+  },
```
Comment (Kamba56) [inline src/redux/books/Books.js:56]: - [x] I noticed that books added or removed do not reflect on the page until I reload the page. Please make sure they display immediately after they are added. You can do this by updating not just the API but also update the store.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @Haywayaheadshot , 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great use of Github workflow!👏👏👏
- [x] All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/Books.js (hunk 1 above)
Comment (Haywayaheadshot) [reply src/redux/books/Books.js:56]: @Kamba56 Thank you for reviewing my project. I honestly tried to do that and couldn't before submitting it. Do you think you can give me a hint as to what to do particularly to achieve this?

**Comment by Haywayaheadshot** [comment]
@Kamba56 Thank you for reviewing my project once again. I have resolved the bug but I did it but increasing the setTimeOut duration to allow the UI update

**Review by emmiiorji — APPROVED** [review]
Hi @Haywayaheadshot, 

# Status: Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats-13

## Good Highlights✔

- Linter checks are passing
- Requested changes have been implemented

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Haywayaheadshot** [comment]
@emmiiorji Thank you for reviewing my project

**Pull request merged**